### PR TITLE
[lldb] Streamline ConstString -> std::string conversion (NFC)

### PR DESCRIPTION
**Pull request by JDevlieghere**
Make it easier to go from a ConstString to a std::string without having to go through a C-String or a llvm::StringRef. I made the conversion operator explicit as this is a relatively expensive operations (compared to a StringRef or string_view).

### CODE REVIEW

**Review by bulbazord — APPROVED** [review]

**Review by medismailben — APPROVED** [review]

File: lldb/include/lldb/Utility/ConstString.h [hunk 1]
```diff
@@ -192,6 +198,9 @@ class ConstString {
     return llvm::StringRef(m_string, GetLength());
   }
 
+  /// Get the string value as a std::string
+  std::string GetString() const { return std::string(m_string, GetLength()); }
```
Comment (felipepiovezan) [inline lldb/include/lldb/Utility/ConstString.h:202]: why is this necessary, if we have the explicit operator?

**Review by felipepiovezan — APPROVED** [review]

File: lldb/include/lldb/Utility/ConstString.h (hunk 1 above)
Comment (JDevlieghere) [reply lldb/include/lldb/Utility/ConstString.h:202]: It's not necessary, it's more for consistency with how we already do things (similar to how `StringRef` has the explicit operator and the `.str()` method on the llvm side).
